eodatasets3/wagl.py

```diff
@@ -350,6 +350,19 @@ def val(v: Any):
         secho("(done)")
 
 
+def _get_level1_metadata_path(wagl_doc):
+    source_level1 = Path(get_path(wagl_doc, ("source_datasets", "source_level1")))
+    # The odc-metadata can be in the source directory or alongside it.
+    if source_level1.exists():
+        level1_metadata_path = source_level1.with_suffix(".odc-metadata.yaml")
+        if not level1_metadata_path.exists():
+            # Assume the yaml is in the source level1 dir
+            [level1_metadata_path] = source_level1.rglob("*.odc-metadata.yaml")
```

Comment:
I'm iffy about this. There can be many level 1 scenes in one directory (as they're usually zip files or tar files), and when a metadata file is missing or deleted, this will quietly grab another arbitrary one in the same directory. Which will be very wrong.

Currently we have a "standard" naming convention of metadata being the identical filename to data, with an .odc-metadata.yaml suffix. And if users want to put their metadata elsewhere, they specify level 1 path explicitly to the packager: `--level1 /some/path/to/yaml`.

Is there are third option you need to solve here?

Reply:
We _really_ don't want to quietly link to the wrong source provenance, so I'm hesitant to make this level1 scanning too loose and magic.

Reply:
Thanks Jeremy.  I've dialled back the magic.